python/lsst/fgcmcal/fgcmBuildStars.py

```diff
@@ -564,23 +443,98 @@ def _fillVisitCatalog(self, butler, visitCat, srcVisits, srcCcds):
 
             rec['psfSigma'] = psf.computeShape().getDeterminantRadius()
 
-            if butler.datasetExists('calexpBackground', dataId=dataId):
+            if dataRef.datasetExists(datasetType='calexpBackground'):
                 # Get background for reference CCD
                 # This approximation is good enough for now
                 bgStats = (bg[0].getStatsImage().getImage().array
-                           for bg in butler.get('calexpBackground',
-                                                dataId=dataId))
+                           for bg in dataRef.get(datasetType='calexpBackground'))
                 rec['skyBackground'] = sum(np.median(bg[np.isfinite(bg)]) for bg in bgStats)
             else:
+                self.log.warn('Sky background not found for visit %d / ccd %d' %
+                              (visit, dataRef.dataId[self.config.ccdDataRefName]))
                 rec['skyBackground'] = -1.0
 
-    def fgcmMakeAllStarObservations(self, butler, visitCat):
+    def findAndGroupDataRefs(self, butler, dataRefs):
         """
-        Compile all good star observations from visits in visitCat
+        Find and group dataRefs (by visit).  If dataRefs is an empty list,
+        this will look for all source catalogs in a given repo.
 
         Parameters
         ----------
         butler: `lsst.daf.persistence.Butler`
+        dataRefs: `list` of `lsst.daf.persistence.ButlerDataRef`
+           Data references for the input visits.
+           If this is an empty list, all visits with src catalogs in
+           the repository are used.
+
+        Returns
+        -------
+        groupedDataRefs: `dict`
+           Dictionary with visit keys, and `list`s of `lsst.daf.persistence.ButlerDataRef`
+        """
+
+        camera = butler.get('camera')
+
+        ccdIds = []
+        for detector in camera:
+            ccdIds.append(detector.getId())
+
+        # TODO: related to DM-13730, this dance of looking for source visits
+        # will be unnecessary with Gen3 Butler.  This should be part of
+        # DM-13730.
+
+        scanAllCcds = False
+        if len(dataRefs) == 0:
+            if not self.config.checkAllCcds:
```

Comment:
I'm confused about `scanAllCcds` vs `checkAllCcds`. It looks like `scanAllCcds` is `True` only if `checkAllCcds` is `False`, but it looks like it is only checking the reference CCD in that case. Should the line `scanAllCcds = True` be moved to the `else:` statement below?